### ISSUE TITLE
LSP completions violate spec

### ISSUE THREAD

**Issue by krobelus** [elves/elvish#1868]
### What happened, and what did you expect to happen?

LSP completions include this, which violates the spec because kind should be either "plaintext" or "markdown".

```
      "documentation": {
        "kind": "",
        "value": ""
      },
```

`documentation` is an optional [markupContent](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#markupContent)

go-lsp package defines it as

```go
	// A human-readable string that represents a doc-comment.
	Documentation MarkupContent `json:"documentation,omitempty"`
```

`omitempty` is useless. To fix this, we could change it to `Documentation *MarkupContent`.

Opening this issue here because it seems like https://github.com/nimblebun/go-lsp is not active.
Maybe we should switch to a different library?

### Output of "elvish -version"

0.21.0

### Code of Conduct

- [X] I agree to follow Elvish's [Code of Conduct](https://src.elv.sh/CODE_OF_CONDUCT.md).

**Comment by xiaq** [comment]
That's right, we should switch to a more actively maintained LSP type package. Any suggestions?

**Comment by krobelus** [comment]
I'm not super well informed but I'd probably start with autogenerated message buffers like [what gopls uses](https://github.com/golang/tools/blob/8843590264d168027205f86e897927c339883bab/gopls/internal/protocol/tsprotocol.go#L5-L11).
Regarding the issue at hand, they define it as pointer, so the zero value will be serialized correctly

```go
{
    Documentation *Or_CompletionItem_documentation `json:"documentation,omitempty"`
}
```
